server/src/domain/person/person.service.ts

```diff
@@ -332,44 +333,115 @@ export class PersonService {
     this.logger.debug(`${faces.length} faces detected in ${asset.resizePath}`);
     this.logger.verbose(faces.map((face) => ({ ...face, embedding: `vector(${face.embedding.length})` })));
 
-    for (const { embedding, ...rest } of faces) {
-      const matches = await this.smartInfoRepository.searchFaces({
-        userIds: [asset.ownerId],
-        embedding,
-        numResults: 1,
-        maxDistance: machineLearning.facialRecognition.maxDistance,
-      });
+    for (const face of faces) {
+      const mappedFace = {
+        assetId: asset.id,
+        embedding: face.embedding,
+        imageHeight: face.imageHeight,
+        imageWidth: face.imageWidth,
+        boundingBoxX1: face.boundingBox.x1,
+        boundingBoxX2: face.boundingBox.x2,
+        boundingBoxY1: face.boundingBox.y1,
+        boundingBoxY2: face.boundingBox.y2,
+      };
+
+      await this.repository.createFace(mappedFace);
+    }
 
-      let personId = matches[0]?.personId || null;
-      let newPerson: PersonEntity | null = null;
-      if (!personId) {
-        this.logger.debug('No matches, creating a new person.');
-        newPerson = await this.repository.create({ ownerId: asset.ownerId });
-        personId = newPerson.id;
-      }
+    await this.assetRepository.upsertJobStatus({
+      assetId: asset.id,
+      facesRecognizedAt: new Date(),
+    });
 
-      const face = await this.repository.createFace({
-        assetId: asset.id,
-        personId,
+    return true;
+  }
+
+  async handleQueueRecognizeFaces({ force }: IBaseJob) {
+    const { machineLearning } = await this.configCore.getConfig();
+    if (!machineLearning.enabled || !machineLearning.facialRecognition.enabled) {
+      return true;
+    }
+
+    await this.jobRepository.waitForQueueCompletion(QueueName.THUMBNAIL_GENERATION, QueueName.FACE_DETECTION);
```

Comment:
I think bullmq will automatically re-queue a job if it takes longer than 30 seconds. I think there is a way to prevent this though.